### ISSUE TITLE
symfony cli not honoring $COMPOSER_HOME

### ISSUE THREAD

**Issue by verfriemelt-dot-org** [symfony-cli/symfony-cli#316]
symfony: 5.5.3
composer: 2.5.5

i defined composer_home as /tmp in my container:

```
easteregg@9f3a07d496be:/var/www$ env | grep -i composer
COMPOSER_NO_INTERACTION=1
COMPOSER_DISCARD_CHANGES=1
COMPOSER_HOME=/tmp
```

which results in this failure:

```
easteregg@9f3a07d496be:/var/www$ symfony new testing
* Creating a new Symfony project with Composer
  (running /usr/local/bin/composer create-project symfony/skeleton /var/www/testing  --no-interaction)

mkdir /home/easteregg: permission denied


  unable to run /usr/local/bin/composer create-project symfony/skeleton /var/www/testing  --no-interaction

```

since `/home/easteregg` is not existing , that is to be expected, hence i set the `COMPOSER_HOME` variable.

if i grab the tried excution, it works just as expected, since - i assume - the env is not dropped.

so we should at least preserve several `COMPOSER_*` vars :thinking:

**Comment by tucksaun** [comment]
As far as I know, we propagate the whole env (see https://github.com/symfony-cli/symfony-cli/blob/6c6a669bfa2a1cf600cc3aff704e58854f24dda7/local/php/executor.go#L322)

So I'm wondering if the error is not coming from the CLI itself when creating temporary files. 
What happens if you run `symfony php -r 'var_dump($_SERVER);'` ?

**Comment by verfriemelt-dot-org** [comment]
actually something completly different :D

```
easteregg@9f3a07d496be:/var/www$ symfony php -r 'var_dump($_SERVER);'
mkdir /home/easteregg: permission denied
```

**Comment by tucksaun** [comment]
Exactly the same `mkdir /home/easteregg: permission denied`.
That confirms my theory.
Can you try again having a valid home directory?

I'm not sure supporting a read-only or non-existent user's home directory for a development tool is something worth supporting  🤔

**Comment by verfriemelt-dot-org** [comment]
i see the point about the invalid home dir for sure :thinking: 
at least a more descriptive error message would be helpful i think, because my direction went to "composer is failing" :thinking: 

depending on the data to be written, it might be possible to add an fallback to `/tmp`.
thanks for your help! :heart:

**Comment by verfriemelt-dot-org** [comment]
just as a followup, after adding a 

```
RUN mkdir /home/$USER && chown $USER:$USER /home/$USER
```
everything works fine.
the `$ symfony php -r 'var_dump($_SERVER);'` command

creates the following tmp folders:

```
root@1c603674239c:/var/www# tree /home/easteregg/ -a
/home/easteregg/
|-- .bash_history
`-- .symfony5
    `-- tmp
        `-- ch5cre7eoq1c755cg51g
```

**Comment by tucksaun** [comment]
Yes this the expected behavior as this is done to ensure any execution of `php` binary (or related binaries) would use the PHP installation detected and selected by the CLI.

**Comment by tucksaun** [comment]
@verfriemelt-dot-org can you confirm we can close this one? thx!

**Comment by tucksaun** [comment]
adding some documentation should be relatively easy for a first contribution
